railties/lib/rails/generators/rails/app/templates/Gemfile.tt

```diff
@@ -56,6 +56,10 @@ group :development do
 <%- end -%>
   # Speed up commands on slow machines / big apps [https://github.com/rails/spring]
   # gem "spring"
+
+<%- if RUBY_VERSION >= "3.1" -%>
+  gem "error_highlight", ">= 0.4.0", platforms: [:ruby]
```

Comment:
Ruby 3.1 ships with `error_highlight` `0.3` right?

Do you need `0.4.0` specifically, or could we just try to load the one shipped with Ruby?

Reply:
I'm guessing we need 0.4.0 for the `backtrace_location` keyword added in https://github.com/ruby/error_highlight/commit/2735e4681ac0890dcc28857a5f74d954bcc9e316

I think that means we should probably guard for that keyword (or alternatively for the `error_highlight` version) above so that uses who happen to have `backtrace_location `0.3.0` don't have issues.

Reply:
@jhawthorn is right. Ruby 3.1 ships error_highlight 0.3.0, which does not have an API for external (I regret not having it), so unfortunately it only works with error_highlight 0.4.0 or later.